app/assets/javascripts/components/subjects_taught.js

```diff
@@ -0,0 +1,34 @@
+"use strict";
+
+document.addEventListener("DOMContentLoaded", function() {
+  var fieldset = document.querySelector("#tslr_claim_eligible_subjects");
+
+  function toggleSubjectValues(event) {
+    var el = event.target;
+    var eligibleSubjectsName = "tslr_claim[eligible_subjects][]";
+    var notTeachingSubjectsName = "tslr_claim[mostly_teaching_eligible_subjects]";
+
+    if (!el) {
+      return;
+    }
+    if (el.checked === false) {
+      return;
+    }
+
+    if (el.name == eligibleSubjectsName) {
+      var radioButton = document.querySelector("input[type='radio'][name='" + notTeachingSubjectsName + "']");
+      radioButton.checked = false;
+    } else if (el.name == notTeachingSubjectsName) {
+      var checkboxes = document.querySelectorAll("input[name='" + eligibleSubjectsName + "']");
+      checkboxes.forEach(function(item, index) {
+        item.checked = false;
+      });
+    }
+  }
+
+  if (!fieldset) {
+    return;
+  }
```

Comment:
Probably want to do this guard at the top, so we can be confident that the rest of the code only runs on the expected pages.